File: Corte1/C++/Parcial/Estudiante.cpp
#ifndef ESTUDIANTE_CPP
#define ESTUDIANTE_CPP
#include <string>

class Estudiante {
private:
    // REQUERIMIENTO 25: Atributos privados (Encapsulamiento)
    std::string codigo, nombre;
    int edad;

public:
    // REQUERIMIENTO 27: Sobrecarga de Constructores
    Estudiante() : codigo(""), nombre(""), edad(0) {} // Por defecto
    
    Estudiante(std::string c, std::string n, int e) // Con parámetros
        : codigo(c), nombre(n), edad(e) {}

    // Getters para acceder a los datos privados
    std::string getCodigo() { return codigo; }
    std::string getNombre() { return nombre; }
};
#endif

File: Corte2/Python/Evaluación/Estudiante.py
class Estudiante:
    # REQUERIMIENTO 25: Atributos privados (Encapsulamiento)
    # REQUERIMIENTO 27: Sobrecarga de Constructores
    # En Python se usa un constructor con valores por defecto para simular sobrecarga
    def __init__(self, c="", n="", e=0):
        self.__codigo = c
        self.__nombre = n
        self.__edad = e

    # Getters para acceder a los datos privados
    def get_codigo(self):
        return self.__codigo

    def get_nombre(self):
        return self.__nombre